security/src/main/java/io/micronaut/security/authentication/BasicAuthAuthenticationFetcher.java

```diff
@@ -65,9 +67,9 @@ public BasicAuthAuthenticationFetcher(Authenticator authenticator,
             Flowable<AuthenticationResponse> authenticationResponse = Flowable.fromPublisher(authenticator.authenticate(request, credentials.get()));
 
             return authenticationResponse.switchMap(response -> {
-                if (response.isAuthenticated()) {
+                if (response.isAuthenticated() && response.getUserDetails().isPresent()) {
```

Comment:
The contract of the API is that if `isAuthenticated` returns true, the user details should be present. So this check is unnecessary

Reply:
I've removed the check.